addons/website_sale/models/sale_order.py

```diff
@@ -511,6 +511,16 @@ def _update_cart_line_values(self, order_line, update_values):
         self.ensure_one()
         order_line.write(update_values)
 
+    def _validate_zero_amount_cart(self):
+        """
+        This method is used to validate the cart when the total amount of cart is zero.
+        This method is intended to be overridden by other modules to extend its functionality
+        such as adding automatic invoicing for zero amount orders
+
+        :return: False, when no invoice is generated for this order and True otherwise
```

Comment:
not returning a bool anymore (and honestly not sure there is a need for a docstring, the code is self-explaining IMHO)